### DISCUSSION TITLE
undefined reference to `csc_matrix'

### DISCUSSION THREAD

**Discussion by Seele13** [osqp/osqp#545] [C Interface] [answered]
I came across it while compiling a project on the ARM platform ——undefined reference to 'csc_matrix'. However, no errors are reported on x86-64 platforms.
![image](https://github.com/osqp/osqp/assets/29163049/76372394-f059-4e16-bea1-fb37e4333893)

**Comment by Seele13** [comment]
I use gcc-linaro-7.5.0-2019.12-x86_64_arm-linux-gnueabihf for arm compilation

**Comment by imciner2** [comment, marked as answer]
What defines/settings are you using to compile the code? Are you using the EMBEDDED flag? Because that function is only enabled in the non-embedded mode since it has memory allocations inside.

**Comment by Seele13** [comment]
I got it. Thanks!!!
